src/webgpu/api/operation/rendering/indirect_draw.spec.ts

```diff
@@ -1,13 +1,184 @@
 export const description = `
 Tests for the indirect-specific aspects of drawIndirect/drawIndexedIndirect.
-
-TODO:
-* parameter_packing - Test that the indirect draw parameters are tightly packed.
-  - offset= {0, 4, k * sizeof(args struct), k * sizeof(args struct) + 4}
-  - mode= {drawIndirect, drawIndexedIndirect}
 `;
 
+import { params, poptions } from '../../../../common/framework/params_builder.js';
 import { makeTestGroup } from '../../../../common/framework/test_group.js';
 import { GPUTest } from '../../../gpu_test.js';
 
 export const g = makeTestGroup(GPUTest);
+
+const filled = new Uint8Array([0, 255, 0, 255]);
+const notFilled = new Uint8Array([0, 0, 0, 0]);
+
+const kDrawIndirectParametersSize = 4 * Uint32Array.BYTES_PER_ELEMENT;
+
+g.test('basics,drawIndirect')
+  .desc(
+    `Test that the indirect draw parameters are tightly packed for drawIndirect.
+An indirectBuffer is created based on indirectOffset. The actual draw args being used indicated by the
+indirectOffset is going to draw a left bottom triangle.
+While the remaining indirectBuffer is populated with random numbers or draw args
+that draw right top triangle, both, or nothing which will fail the color check.
+The test will check render target to see if only the left bottom area is filled,
+meaning the expected draw args is uploaded correctly by the indirectBuffer and indirectOffset.
+
+Params:
+    - indirectOffset= {0, 4, k * sizeof(args struct), k * sizeof(args struct) + 4}
+    `
+  )
+  .cases(
```

Comment:
nit: I think these should all be subcases.
```suggestion
  .subcases(() =>
```

Reply:
Do we have some docs for these helpers `cases`, `subcases`, `poptions` ... (or are they names from some common practice?)
I still find a bit confusing to read their source code 😅 
(I now get subcases are without extra dropdown menu)

Reply:
Probably not, but great idea. I will add some docs.